### PR TITLE
[BigQueryWorkloadTester] Adding BigQuery backend and test

### PR DESCRIPTION
**Pull request by ldanielmadariaga**

### CODE REVIEW

File: BigQueryWorkloadTester/src/main/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendFactory.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+public class BigQueryBackendFactory {
+
+  private static final Logger logger = Logger.getLogger(BigQueryBackendFactory.class.getName());
+
+  /**
+   * Ensures that the BigQueryBackend and it's dependencies are properly built and configured.
+   *
+   * @return a properly configured BigQueryBackend
+   */
+  public static BigQueryBackend get(
```
Comment (EricBeach) [inline BigQueryWorkloadTester/src/main/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendFactory.java]: can we have a more descriptive name than get ?

File: BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+
+/** Tests for {@link BigQueryBackend}. */
+@RunWith(JUnit4.class)
+public class BigQueryBackendTest {
+  // private static final long START_TIME = 10;
```
Comment (EricBeach) [inline BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java]: ?

File: BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  // private static final long END_TIME = 100;
```
Comment (EricBeach) [inline BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java]: ?

File: BigQueryWorkloadTester/src/main/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendFactory.java (hunk 1 above)
Comment (ldanielmadariaga) [reply BigQueryWorkloadTester/src/main/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendFactory.java]: Done

File: BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java (hunk 2 above)
Comment (ldanielmadariaga) [reply BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java]: Sorry refactored this and missed deleting the original variables

File: BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java (hunk 3 above)
Comment (ldanielmadariaga) [reply BigQueryWorkloadTester/src/test/java/com/google/cloud/pontem/benchmark/backends/BigQueryBackendTest.java]: Sorry refactored this and missed deleting the original variables

**Pull request merged**